### PR TITLE
Remove unneeded node groups regardless of scale down being in cooldown.

### PR DESCRIPTION
**Pull request by olagacek**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Currently unneeded node groups are deleted only if scale down isn't in cooldown, so in particular if there was a scale up in a given loop, no unneeded node group will get deleted. There are certain scenarios in which that might turn out problematic - eg if scale up fails in every loop, the node groups won't be deleted for a long time. At the same time, unneeded node groups might affect scale up failures (e.g. quota issues).

Due to that, we can decouple unneeded node group deletion from the rest of scale down logic. 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cluster-autoscaler/core/static_autoscaler.go [hunk 1]
```diff
@@ -621,23 +621,27 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) caerrors.AutoscalerErr
...
+		// in progress.
+		_, drained := scaleDownActuationStatus.DeletionsInProgress()
+		var removedNodeGroups []cloudprovider.NodeGroup
+		if len(drained) == 0 {
+			var err error
+			removedNodeGroups, err = a.processors.NodeGroupManager.RemoveUnneededNodeGroups(autoscalingContext)
+			if err != nil {
+				klog.Errorf("Error while removing unneeded node groups: %v", err)
+			}
+			scaleDownStatus.RemovedNodeGroups = removedNodeGroups
```
Comment (x13n) [inline cluster-autoscaler/core/static_autoscaler.go:634]: I think this makes line 653 no-op now.

**Review by x13n — COMMENTED** [review]
/assign

Looks good, just one comment.

File: cluster-autoscaler/core/static_autoscaler.go (hunk 1 above)
Comment (olagacek) [reply cluster-autoscaler/core/static_autoscaler.go:634]: I don't think so, as in line 648 we are overwriting `scaleDownStatus`

Comment (x13n) [reply cluster-autoscaler/core/static_autoscaler.go:634]: Ah, good point. This function is way too long :disappointed:

**Pull request merged**